scalableminds/webknossos
# [WIP] Add perspective camera for td viewport

This pr tries to add a perspective camera to the 3D viewport. More information is posted below ⬇️ 



### URL of deployed dev instance (used for testing):
- https://addperspectivecamerafortd.webknossos.xyz

### Steps to test:
- visit https://addperspectivecamerafortd.webknossos.xyz/datasets/sample_organization/l4_sample/view#3584,3584,1024,0,1.3
- (log in)
- use the 3D viewport and ensure that the perspective camera setting is enabled (see screenshot)
![image](https://user-images.githubusercontent.com/2486553/183040425-18bc2761-d2f3-4b46-b91a-2dcb03007e1f.png)
- pan or rotate with left- or right-click drag
- both panning and rotating should behave natural, but feel buggy currently (the rotation center should always be the intersection of the visible planes; see screenshot)
![image](https://user-images.githubusercontent.com/2486553/183041055-82d7cc69-fa09-4c48-94b5-648d6e68d4de.png)
- zoom in/out is done via mouse wheel

### Issues:
- fixes #6260

------
(Please delete unneeded items, merge only when none are left open)
-----

![image](https://user-images.githubusercontent.com/1390035/181256774-db70b65b-f087-4180-b879-af0962dbe9db.png)
